ProcessLib/CoupledSolutionsForStaggeredScheme.cpp

```diff
@@ -27,48 +29,33 @@ CoupledSolutionsForStaggeredScheme::CoupledSolutionsForStaggeredScheme(
     }
 }
 
-std::vector<std::vector<double>> getPreviousLocalSolutions(
-    const CoupledSolutionsForStaggeredScheme& cpl_xs,
-    const std::vector<std::vector<GlobalIndexType>>& indices)
+std::vector<double> getCoupledLocalSolutions(
+    std::vector<GlobalVector*> const& global_solutions,
+    std::vector<std::vector<GlobalIndexType>> const& indices)
 {
-    if (cpl_xs.coupled_xs_t0.empty())
+    if (global_solutions.empty())
     {
         return {};
     }
 
-    const auto number_of_coupled_solutions = cpl_xs.coupled_xs.size();
-    std::vector<std::vector<double>> local_xs_t0;
-    local_xs_t0.reserve(number_of_coupled_solutions);
-
-    int coupling_id = 0;
-    for (auto const& x_t0 : cpl_xs.coupled_xs_t0)
+    std::size_t const local_solutions_size = std::accumulate(
+        cbegin(indices),
+        cend(indices),
+        std::size_t(0),
+        [](GlobalIndexType const size,
+           std::vector<GlobalIndexType> const& process_indices) {
+            return size + process_indices.size();
+        });
+    std::vector<double> result;
+    result.reserve(local_solutions_size);
+
+    int process_id = 0;
+    for (auto const& x : global_solutions)
     {
-        local_xs_t0.emplace_back(x_t0->get(indices[coupling_id]));
-        coupling_id++;
+        auto const& values = x->get(indices[process_id]);
+        result.insert(cend(result), cbegin(values), cend(values));
+        process_id++;
     }
```

Comment:
How about writing as follows,
```
std::vector<double> local_solutions;
local_solutions.reserve(local_solution_size);

for (int process_id = 0; process_id < static_cast<int>(global_solutions); ++process_id)
{
    auto const& local_solution = global_solutions[process_id]->get(indices[process_id]);
    std::copy(local_solution.begin(), local_solution.end(), std::back_inserter(local_solutions));
}
```

Reply:
I see three suggestions:
 - rename result to `local_solutions`. That's easy.
 - use process_id as loop counter. Will think about it.
 - use copy not insert. Copy is slower because of push-back one-by-one. Insert works (in this case) on a chunk. But another thing can be done, is to use `std::make_move_iterator`.
Will push the changes in a minute.